operator/pkg/controlplane/metricsadapter/mainfests.go

```diff
@@ -0,0 +1,95 @@
+package metricsadapter
+
+const (
+	// KarmadaMetricsAdapterDeployment is karmada-metrics-adapter deployment manifest
+	KarmadaMetricsAdapterDeployment = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .DeploymentName }}
+  namespace: {{ .Namespace }}
+  labels:
+    karmada-app: karmada-metrics-adapter
+    app.kubernetes.io/managed-by: karmada-operator
+spec:
+  replicas: {{ .Replicas }}
+  selector:
+    matchLabels:
+      karmada-app: karmada-metrics-adapter
+  template:
+    metadata:
+      labels:
+        karmada-app: karmada-metrics-adapter
+    spec:
+      automountServiceAccountToken: false
+      tolerations:
+        - key: node-role.kubernetes.io/master
+          operator: Exists
+      containers:
+      - name: karmada-metrics-adapter
+        image: {{ .Image }}
+        imagePullPolicy: IfNotPresent
+        command:
+        - /bin/karmada-metrics-adapter
+        - --kubeconfig=/etc/karmada/config
+        - --authentication-kubeconfig=/etc/karmada/config
+        - --authorization-kubeconfig=/etc/karmada/config
+        - --client-ca-file=/etc/karmada/pki/ca.crt
+        - --audit-log-path=-
+        - --audit-log-maxage=0
+        - --audit-log-maxbackup=0
+        volumeMounts:
+        - name: kubeconfig
+          subPath: config
+          mountPath: /etc/karmada/config
+        - name: karmada-cert
+          mountPath: /etc/karmada/pki
+          readOnly: true
+        readinessProbe:
+          httpGet:
+            path: /readyz
+            port: 443
+            scheme: HTTPS
+          initialDelaySeconds: 1
+          failureThreshold: 3
+          periodSeconds: 3
+          timeoutSeconds: 15
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 443
+            scheme: HTTPS
+          initialDelaySeconds: 10
+          failureThreshold: 3
+          periodSeconds: 10
+          timeoutSeconds: 15
+        resources:
+          requests:
+            cpu: 100m
```

Comment:
I noticed that all the components only specify CPU requests. Should we also include memory requests?  @XiShanYongYe-Chang 
We need to set this configuration for the production environment.

Reply:
What do you think? @calvin0327

Reply:
I thinks the real request of CUP and memery of karmada components is hard to get an exact value.

Reply:
Just have a general test: https://karmada.io/zh/blog/2022/10/26/test-report/

How about setting the CPU to 500m and the memory to 500Mi? Let's make sure it works on a small scale.
cc @RainbowMango for advice

Reply:
I'm not sure what the `small scale` is, maybe we can start by not setting the required resources. I noticed the other components don't have such a configuration, such as [karmada-controller-management](https://github.com/karmada-io/karmada/blob/276d6a8ad1f156546a3208aefb26ceb3252c3a7f/operator/pkg/controlplane/manifests.go#L85C2-L85C37).

This is a good question though, we can consider allowing users to set them from API, after that, we can set a minium value then.

Reply:
@RainbowMango @jwcesign  There is corresponding struct In [Karmada API](https://github.com/karmada-io/karmada/blob/master/operator/pkg/apis/operator/v1alpha1/type.go#L518), but now, we don't implement this feature. Can I do it in a new pr?

Reply:
Sure.